sdks/python/apache_beam/internal/apiclient.py

```diff
@@ -242,12 +243,28 @@ def __init__(self, packages, options, environment_version):
           dataflow.Environment.SdkPipelineOptionsValue.AdditionalProperty(
               key='options', value=to_json_value(options_dict)))
 
-      dd = DisplayData.create_from(options)
-      items = [item.get_dict() for item in dd.items]
+      items = self.format_options_display_data(options)
       self.proto.sdkPipelineOptions.additionalProperties.append(
           dataflow.Environment.SdkPipelineOptionsValue.AdditionalProperty(
               key='display_data', value=to_json_value(items)))
 
+  @classmethod
+  def format_options_display_data(cls, options):
```

Comment:
Why don't you do this at DisplayDataItem.__init__? if the value is an instance of list, convert value first and set the type based on converted value.